avocado/plugins/assets.py

```diff
@@ -265,10 +265,14 @@ def pre_tests(self, job):
                 # runnable, because the requirements resolution planned is
                 # completely different from the traditional job runner
                 if isinstance(test, Runnable):
-                    continue
+                    if test.kind == 'avocado-instrumented':
+                        module_path, klass_method = test.uri.split(':', 1)
+                        klass, method = klass_method.split('.', 1)
+                        candidate = (module_path, klass, method)
+                        candidates.append(candidate)
 
                 # fetch assets only on instrumented tests
-                if isinstance(test[0], str):
+                elif isinstance(test[0], str):
```

Comment:
IMO, this `elif` is not necessary and can be removed because the instrumented tests are already resolved.

Reply:
Maybe I'm missing something about your comment.  AFAICT this *remains* necessary, because without it, tests resolved via the loader won't be processed by the plugin, making it ineffective.

You can check that (I just did to be sure) by removing the entire `elif` block and revert the tests to use the legacy runner, and you'll see the failures.

Now if you meant removing just the `elif` *check*, that is, `isinstance(test[0], str)`, then it'd require a `continue` after the end of the `if  `isinstance(test, Runnable)` block.

Reply:
Right.  But to avoid breakages in between those commits (which would mess up bisection) that removal needs to happen either at the removal of the `runner` (where users would have a hard time to reach that code) or even at the removal of the `loader` code (where then it would be impossible to have this type of test suite created).

If you don't mind, I'll remove it together with the loader code (hopefully very very shortly).

Reply:
Sure, I don't have a problem with removal together with the loader code. I am just a little afraid that we will forget about it.

Reply:
OK then.  I already included that in my WIP removal of the loader.  I'll close this PR in favor of #5362 then.

Thanks for the review!